android-runtime/runtime/src/main/java/io/sdkgen/runtime/SdkgenHttpClient.kt

```diff
@@ -0,0 +1,244 @@
+                addProperty("width", size.x)
+                addProperty("height", size.y)
+            })
+        }
+    }
+
+    @SuppressLint("HardwareIds")
+    suspend fun makeRequest(
+        functionName: String,
+        bodyArgs: JsonObject?,
+        timeoutMillis: Long? = null
+    ): InternalResponse = suspendCoroutine { continuation ->
+        try {
+            val body = JsonObject().apply {
+                addProperty("version", 3)
+                addProperty("requestId", callId())
+                addProperty("name", functionName)
+                add("args", bodyArgs ?: JsonObject())
+                add("extra", JsonObject().apply { addProperty("fingerprint", Settings.Secure.getString(applicationContext.contentResolver, Settings.Secure.ANDROID_ID)) })
+                add("deviceInfo", makeDeviceObj())
+            }
+
+            val request = Request.Builder()
+                .url("$baseUrl${if (baseUrl.last() == '/') "" else "/"}$functionName")
+                .post(body.toString().toRequestBody("application/json; charset=utf-8".toMediaType()))
+                .build()
+
+            val call = httpClient.newCall(request)
+            call.timeout().timeout(timeoutMillis ?: defaultTimeoutMillis, TimeUnit.MILLISECONDS)
+
+            try {
+                val response = call.execute()
+
+                if (response.code in 501..599) {
+                    continuation.resume(
+                        InternalResponse(
+                            JsonObject().apply {
+                                addProperty("type", "Fatal")
+                                addProperty("message", applicationContext.getString(R.string.sdkgen_error_call_code, response.code.toString()))
+                            },
+                            null
+                        )
+                    )
+
+                    return@suspendCoroutine
+                }
+
+                val responseBody = try {
+                    val stringBody = response.body?.string()
+                    gson.fromJson(stringBody, JsonObject::class.java)
+                } catch (e: Exception) {
+                    e.printStackTrace()
+
+                    continuation.resume(
+                        InternalResponse(
+                            JsonObject().apply {
+                                addProperty("type", "Fatal")
+                                addProperty("message", applicationContext.getString(R.string.sdkgen_error_serialization))
+                            },
+                            null
+                        )
+                    )
+
+                    return@suspendCoroutine
+                }
+
+                if (response.code != 200) {
+                    val jsonError = responseBody.getAsJsonObject("error")
+                    continuation.resume(InternalResponse(jsonError, null))
+                } else {
+                    continuation.resume(InternalResponse(null, responseBody))
```

Comment:
only `result` field of `responseBody` should passed on, no?

Reply:
the error object is outside the result object, and I needed on the client, as the serialization is done there.